specification/botservice/resource-manager/Microsoft.BotService/stable/2017-12-01/botservice.json

```diff
@@ -392,242 +434,1239 @@
+          },
           {
             "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
           }
         ],
         "responses": {
           "200": {
-            "description": "OK.",
+            "description": "The resource provider should return 200 (OK) to indicate that the operation completed successfully. ",
             "schema": {
-              "$ref": "#/definitions/OperationEntityListResult"
+              "$ref": "#/definitions/BotChannel"
+            }
+          },
+          "201": {
+            "description": "If resource is updated successfully, the service should return 201 (Created). Execution to continue asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/BotChannel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed. If the resource group *or* resource does not exist, 404 (NotFound) should be returned.",
+            "schema": {
+              "$ref": "#/definitions/Error"
             }
           }
-        },
-        "deprecated": false,
-        "x-ms-pageable": {
-          "nextLinkName": "nextLink"
         }
-      }
-    }
-  },
-  "definitions": {
-    "Resource": {
-      "description": "Azure resource",
-      "x-ms-azure-resource": true,
-      "properties": {
-        "id": {
-          "description": "Specifies the resource ID.",
-          "type": "string",
-          "readOnly": true
-        },
-        "name": {
-          "description": "Specifies the name of the resource.",
-          "type": "string",
-          "readOnly": true
-        },
-        "location": {
-          "description": "Specifies the location of the resource.",
-          "type": "string"
+      },
+      "delete": {
+        "tags": [
+          "Channel"
+        ],
+        "description": "Deletes a Channel registration from a Bot Service",
+        "operationId": "Channels_Delete",
+        "x-ms-examples": {
+          "Delete Bot": {
+            "$ref": "./examples/DeleteChannel.json"
+          }
         },
-        "type": {
-          "description": "Specifies the type of the resource.",
-          "type": "string",
-          "readOnly": true
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group within the user's subscription."
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 64,
+            "minLength": 2,
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9_.-]*$",
+            "description": "The name of the Bot resource."
+          },
+          {
+            "name": "channelName",

```

Comment:
It is unclear to me if this is the same as the other "channelName" parameters. Is this supposed to be the same enum as is declared a few times above?

Reply:
I'm centralizing this into a common parameter.